### PR TITLE
Rss Acc Transformation

### PR DESCRIPTION
**Pull request by klemense1**
* Serialization of Acceleration Limits in both street and vehicle coordinates

### CODE REVIEW

File: bark/commons/tests/frenet_test.cc [hunk 1]
```diff
@@ -125,4 +126,83 @@ TEST(frenet_state_two_way, straight_line_top_right) {
...
+      delta_time * last_state(StateDefinition::VEL_POSITION), 0, 0, 1;
+
+  FrenetState last_frenet_state(last_state, line);
+  FrenetState current_frenet_state(current_state, line);
+
+  double acc_lat = LatAccStreetToVehicleCs(acc_lat_street, acc_lon, delta_time,
+                                           current_state, current_frenet_state,
+                                           last_frenet_state);
+
+  EXPECT_EQ(acc_lat, acc_lat_street);
```
Comment (patrickhart) [inline bark/commons/tests/frenet_test.cc:153]: straight line on x. vehicle is driving in x.
Why should there be an acc_lat of 0.1?

File: bark/commons/tests/frenet_test.cc [hunk 2]
```diff
@@ -125,4 +126,83 @@ TEST(frenet_state_two_way, straight_line_top_right) {
...
+      delta_time * last_state(StateDefinition::VEL_POSITION), 0, 0, 1;
+
+  FrenetState last_frenet_state(last_state, line);
+  FrenetState current_frenet_state(current_state, line);
+
+  double acc_lat = LatAccStreetToVehicleCs(acc_lat_street, acc_lon, delta_time,
+                                           current_state, current_frenet_state,
+                                           last_frenet_state);
+
+  EXPECT_GT(acc_lat, acc_lat_street);
```
Comment (patrickhart) [inline bark/commons/tests/frenet_test.cc:180]: why is the acc_lat also 0.1 here?

File: bark/commons/tests/frenet_test.cc [hunk 3]
```diff
@@ -125,4 +126,83 @@ TEST(frenet_state_two_way, straight_line_top_right) {
...
+      delta_time * last_state(StateDefinition::VEL_POSITION), 0, 0, 1;
+
+  FrenetState last_frenet_state(last_state, line);
+  FrenetState current_frenet_state(current_state, line);
+
+  double acc_lat = LatAccStreetToVehicleCs(acc_lat_street, acc_lon, delta_time,
+                                           current_state, current_frenet_state,
+                                           last_frenet_state);
+
+  EXPECT_LT(acc_lat, acc_lat_street);
```
Comment (patrickhart) [inline bark/commons/tests/frenet_test.cc:207]: why is the acc_lat also 0.1 here? should it not be reversed to the left_curved test?

File: bark/commons/transformation/frenet_state.cpp [hunk 4]
```diff
@@ -67,7 +67,8 @@ State FrenetStateToDynamicState(const FrenetState& frenet_state,
 
   // calculate angle from frenet velocities
   //  std::cout << frenet_state.vlat << ", " << frenet_state.vlon << std::endl;
-  const auto angle = mg::NormToPI(atan2(frenet_state.vlat, frenet_state.vlon) + line_angle);  // todo che
+  const auto angle = mg::NormToPI(atan2(frenet_state.vlat, frenet_state.vlon) +
```
Comment (patrickhart) [inline bark/commons/transformation/frenet_state.cpp:70]: this variable seems to be unused?

File: bark/commons/transformation/frenet_state.cpp [hunk 5]
```diff
@@ -82,6 +83,31 @@ State FrenetStateToDynamicState(const FrenetState& frenet_state,
   return state;
 }
 
+double LatAccStreetToVehicleCs(double acc_lat_street, double acc_lon,
```
Comment (patrickhart) [inline bark/commons/transformation/frenet_state.cpp:86]: maybe docstring. What is Cs?

File: bark/models/behavior/behavior_rss/behavior_rss.cpp [hunk 6]
```diff
@@ -45,6 +45,11 @@ Trajectory BehaviorRSSConformant::Plan(
     }
   }
 
+  if (last_state_(StateDefinition::TIME_POSITION) < 0) {
```
Comment (patrickhart) [inline bark/models/behavior/behavior_rss/behavior_rss.cpp:48]: why should the time position ever be smaller than 0? I cannot see that happening

File: bark/models/behavior/behavior_rss/behavior_rss.cpp [hunk 7]
```diff
@@ -117,16 +122,80 @@ Trajectory BehaviorRSSConformant::Plan(
...
+  acc_lim_vehicle_cs.lon_acc_min = rss_rest.longitudinalRange.minimum;
+
+  acc_lim_street_cs.lon_acc_max = rss_rest.longitudinalRange.maximum;
+  acc_lim_street_cs.lon_acc_min = rss_rest.longitudinalRange.minimum;
+  VLOG(4) << "RSS Acceleration Restrictions in Road System: " << rss_rest;
+  VLOG(4) << "Acceleration Limits in Road System: " << acc_lim_street_cs;
+  VLOG(4) << "Acceleration Limits in Vehicle System: " << acc_lim_vehicle_cs;
+
+  SetAccelerationLimitsVehicleCs(acc_lim_vehicle_cs);
+  SetAccelerationLimitsStreetCs(acc_lim_street_cs);
```
Comment (patrickhart) [inline bark/models/behavior/behavior_rss/behavior_rss.cpp:198]: which ones are we using for the restriction actually?

Comment (patrickhart) [reply bark/models/behavior/behavior_rss/behavior_rss.cpp:198]: This function is a bit long, but okay.

File: bark/models/behavior/behavior_rss/behavior_rss.hpp [hunk 8]
```diff
@@ -161,9 +182,11 @@ class BehaviorRSSConformant : public BehaviorModel {
   std::shared_ptr<BaseEvaluator> rss_evaluator_;
   BehaviorRSSConformantStatus behavior_rss_status_;
   double world_time_of_last_rss_violation_;
+  State last_state_;
```
Comment (patrickhart) [inline bark/models/behavior/behavior_rss/behavior_rss.hpp:185]: why do we set a last_state_here? should that not be stored in the agent's history?

I think this also leads to errors, such as checking if the time is valid. In the observed world we should get the agent's state.

**Review by patrickhart — CHANGES_REQUESTED** [review]

File: bark/commons/tests/frenet_test.cc (hunk 1 above)
Comment (klemense1) [reply bark/commons/tests/frenet_test.cc:153]: I assume this to come from RSS. just some random limit...

**Review by patrickhart — APPROVED** [review]

**Comment by patrickhart** [comment]
Integrated in branch `rss_benchmark`.

**Pull request closed without merging**